pkg/cmd/secrets.go

```diff
@@ -538,8 +538,29 @@ func substituteSecrets(cmd *cobra.Command, args []string) {
 		}
 	}
 
+	// parse template file
+	templateBody := controllers.ReadTemplateFile(args[0])
+	template, e := controllers.ParseSecretsTemplate(templateBody)
+	if !e.IsNil() {
+		utils.HandleError(e.Unwrap(), e.Message)
+	}
+
+	fetchAllSecrets := utils.GetBoolFlag(cmd, "fetch-all")
+	var secretNames []string
+	if !fetchAllSecrets {
+		// parse secrets from template
+		for _, field := range controllers.TemplateFields(template) {
+			if strings.HasPrefix(field, "{{.") && strings.HasSuffix(field, "}}") {
+				f := strings.TrimPrefix(field, "{{.")
+				f = strings.TrimSuffix(f, "}}")
+				secretNames = append(secretNames, f)
+			}
+		}
+	}
+
+	// fetch secrets
 	dynamicSecretsTTL := utils.GetDurationFlag(cmd, "dynamic-ttl")
-	response, responseErr := http.GetSecrets(localConfig.APIHost.Value, utils.GetBool(localConfig.VerifyTLS.Value, true), localConfig.Token.Value, localConfig.EnclaveProject.Value, localConfig.EnclaveConfig.Value, nil, true, dynamicSecretsTTL)
+	response, responseErr := http.GetSecrets(localConfig.APIHost.Value, utils.GetBool(localConfig.VerifyTLS.Value, true), localConfig.Token.Value, localConfig.EnclaveProject.Value, localConfig.EnclaveConfig.Value, secretNames, true, dynamicSecretsTTL)
```

Comment:
As it stands, if `secretNames` is `[]` (i.e. no names are found in the template), `GetSecrets` will request all secrets

Reply:
Good call out - though thankfully no longer relevant with the dropping of template parsing functionality.